### PR TITLE
Support data source aliasing using clone

### PR DESCRIPTION
**Pull request by VenelinMartinov**
This changes allows us to support automatic aliasing of legacy data sources. This now means that data sources mimic what we do for resources.

fixes https://github.com/pulumi/pulumi-terraform-bridge/issues/2585
related to https://github.com/pulumi/pulumi-terraform-bridge/pull/1938

### CODE REVIEW

File: pkg/pf/tests/tfgen_test.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ import (
 )
 
 func TestRenameResourceWithAliasInAugmentedProvider(t *testing.T) {
-    t.Parallel()
+	t.Parallel()
```
Comment (iwahbe) [inline pkg/pf/tests/tfgen_test.go:36]: Any idea how this was checked in originally? Do we not lint these files?

File: pkg/pf/tests/tfgen_test.go [hunk 2]
```diff
@@ -63,3 +63,36 @@ func TestRenameResourceWithAliasInAugmentedProvider(t *testing.T) {
 	_, err = i.P.(*muxer.ProviderShim).ResolveDispatch(&i)
 	require.NoError(t, err)
 }
+
+func TestRenameDataSourceWithAliasInAugmentedProvider(t *testing.T) {
```
Comment (iwahbe) [inline pkg/pf/tests/tfgen_test.go:67]: I think we need a test that shows that old and new dispatch actually resolve correctly, (from the Pulumi level), not just that the muxer doesn't fail to resolve.

Also, the text only applies to muxed datasources, so lets put that in the title.
```suggestion
func TestRenameMuxedDataSourceWithAliasInAugmentedProvider(t *testing.T) {
```

File: pkg/tfbridge/info/rename.go [hunk 3]
```diff
@@ -84,7 +84,8 @@ func (p *Provider) RenameDataSource(resourceName string, legacyTok tokens.Module
 			currentInfo.Tok.Name().String()))
 	p.DataSources[resourceName] = &currentInfo
 	p.DataSources[legacyResourceName] = &legacyInfo
-	p.P.DataSourcesMap().Set(legacyResourceName, p.P.DataSourcesMap().Get(resourceName))
+	err := shim.CloneResource(p.P.DataSourcesMap(), resourceName, legacyResourceName)
```
Comment (iwahbe) [inline pkg/tfbridge/info/rename.go:87]: Now would be a great opportunity to add a comment to `RenameDataSource` that explains what it actually does (and fix `resourceName` to `datasourceName`).

File: pkg/pf/tests/tfgen_test.go (hunk 1 above)
Comment (VenelinMartinov) [reply pkg/pf/tests/tfgen_test.go:36]: Yup we skip these: https://github.com/pulumi/pulumi-terraform-bridge/blob/2a8b877dbe500f33980524c68855f0e732c3d6aa/.golangci.yml#L41

**Review by iwahbe — APPROVED** [review]

**Pull request merged**